pkg/restapi/device/deviceapi.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package device
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+
+	"github.com/duo-labs/webauthn/protocol"
+	"github.com/duo-labs/webauthn/webauthn"
+	"github.com/trustbloc/edge-agent/pkg/restapi/common/store/user"
+)
+
+// Device represents the user model
+type Device struct {
+	id          uint64
+	name        string
+	displayName string
+	credentials []webauthn.Credential
+}
+
+// NewDevice creates and returns a new Device
+func NewDevice(user *user.User) *Device {
+	device := &Device{}
+	device.id = randomUint64()
+	device.name = user.Name
+	device.displayName =user.FamilyName
+	device.credentials = []webauthn.Credential{}
+
+	return device
+}
+
+func randomUint64() uint64 {
+	buf := make([]byte, 8)
+	rand.Read(buf)
+	return binary.LittleEndian.Uint64(buf)
+}
+
+// WebAuthnID returns the user's ID
+func (d Device) WebAuthnID() []byte {
+	buf := make([]byte, binary.MaxVarintLen64)
+	binary.PutUvarint(buf, uint64(d.id))
+	return buf
+}
+
+// WebAuthnName returns the user's username
+func (d Device) WebAuthnName() string {
+	return d.name
+}
+
+// WebAuthnDisplayName returns the user's display name
+func (d Device) WebAuthnDisplayName() string {
+	return d.displayName
+}
+
+// WebAuthnIcon is not (yet) implemented
+func (d Device) WebAuthnIcon() string {
+	return ""
+}
+
+// AddCredential associates the credential to the user
+func (d *Device) AddCredential(cred webauthn.Credential) {
+	d.credentials = append(d.credentials, cred)
+}
+
+// WebAuthnCredentials returns credentials owned by the user
+func (d Device) WebAuthnCredentials() []webauthn.Credential {
+	return d.credentials
+}
+
+// CredentialExcludeList returns a CredentialDescriptor array filled
+// with all the user's credentials
+func (d Device) CredentialExcludeList() []protocol.CredentialDescriptor {
```

Comment:
This should be included in the `PublicKeyCredentialCreationOptions`.

Reply:
done

Reply:
done